README-gradle.md

```diff
@@ -0,0 +1,98 @@
+## Using iDFlakies on a Gradle project
+
+### Install iDFlakies and testrunner 
+
+Go to the testrunner root directory, and run: 
+
+```
+mvn clean install
+```
+
+Then go to iDFlakies root directory, and run: 
+
+```
+mvn clean install
+```
+
+### Automatically setting up the build.gradle for iDFlakies
+
+Run the following command to  automatically setup the build.gradle for iDFlakies.
+
+```
+bash gradle_modify/modify_gradle.sh path_to_gradle_project
+```
+
+### Manually setting up the build.gradle for iDFlakies
+
+We can configure the build.gradle file manually instead. We need to follow the rule listed below to successfully set up the project. 
+
+1. First we need to add the following dependencies and repository to the `buildscript{}` in build.gradle
+
+   ```groovy
+   buildscript {
+      repositories {
+         ...
+         mavenLocal()
+         ...
+      }
+   
+      dependencies {
+         ...
+         classpath group: 'edu.illinois.cs',
+         name: 'idflakies', 
+         version: '1.1.0'
+         
+         classpath group: 'edu.illinois.cs',
+         name: 'testrunner-gradle-plugin', 
+         version: '1.2'
+         ...
+      }
+   }
+   ```
+
+2. Then, add `apply plugin: 'testrunner'` to the build.gradle file. The location of this code snippet depends on whether build.gradle file has `allprojects{}` and `subprojects{}`. 
+
+   * If there are both `allprojects{}` and `subprojects{}` methods in gradle.build, then the `apply plugin: "testrunner"` command should be added to the end of  `subprojects{}` and the root (outside of all methods in the build file).
+   * If only `allprojects{}`method is in gradle.build, the command is added to  `allprojects{}` only
+   * If only `subprojects{}` method is in gradle.build, the command is added to both the end of `subprojects{}` and the root. 
+   * If there's no `allprojects{}` and `subprojects{}` in gradle.build, only add the command to the root.
+
+   Here is an example where this code snippet is added outside of all methods.
+
+   ```groovy
+   buildscript {
+      repositories {
+         ...
+         mavenLocal()
+         ...
+      }
+   
+      dependencies {
+         ...
+         classpath group: 'edu.illinois.cs',
+         name: 'idflakies', 
+         version: '1.1.0'
+         
+         classpath group: 'edu.illinois.cs',
+         name: 'testrunner-gradle-plugin', 
+         version: '1.2'
+         ...
+      }
+   }
+   
+   apply plugin: 'testrunner'
+   ```
+3. For each subproject with build.gradle file in its directory, append `apply plugin: 'testrunner'` to the end of the build.gradle in the subproject. 
```

Comment:
(1) Just to be clear, all subprojects with build.gradle files will need to be modified as well right?

(2) If (1) is true, then does the script automatically modify all subprojects' build.gradle files?

Reply:
(1) Yes, all subprojects with build.gradle will need to be modified. 
(2) Yes, the script would automatically modify all of them.